### PR TITLE
Simplify SharedBlobCacheService read path some more

### PR DESCRIPTION
**Pull request by original-brownbear**
No point in doing the map lookup for the executor (this one actually shows in profiling) unless we actually for off, also inlining the listener creation to make this code easier to read and make the diff for further optimizations smaller.

### CODE REVIEW

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/eql-correctness

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/part-2

**Comment by original-brownbear** [comment]
Jenkins test this

File: x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java [hunk 1]
```diff
@@ -781,9 +797,10 @@ void populateAndRead(
             }
         }
 
-        private void fillGaps(RangeMissingHandler writer, Executor executor, SharedBytes.IO fileChannel, List<SparseFileTracker.Gap> gaps) {
+        private void fillGaps(RangeMissingHandler writer, String executor, SharedBytes.IO fileChannel, List<SparseFileTracker.Gap> gaps) {
+            final var exec = threadPool.executor(executor);
```
Comment (DaveCTurner) [inline x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java:801]: Can we just do this lookup once up-front, hold the relevant ones in fields on `SharedBlobCacheService`, which is then accessible via the `CacheFile` so we can just pass the actual `Executor` all the way down the stack?

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Pull request merged**